control/canonical.py

```diff
@@ -36,6 +36,28 @@ def canonical_form(xsys, form='reachable'):
         System in desired canonical form, with state 'z'
     T : (M, M) real ndarray
         Coordinate transformation matrix, z = T * x
+
+    Examples
+    --------
+    >>> from control import canonical_form, tf, tf2ss
+
+    >>> G = tf([1],[1, 3, 2])
+    >>> Gs = tf2ss(G)
+
+    >>> Gc, T = canonical_form(Gs)  # default reachable
+    >>> Gc.B                                                    # doctest: +SKIP
+    matrix([[1.],
+            [0.]])
```

Comment:
I fixed these in [98fdb35](https://github.com/python-control/python-control/pull/868/commits/98fdb35f06d0ab02d61747e8f807dd1fa6ba842c).  In a few cases I had to use `round()` to get rid of problems between platforms (see [1d1c289](https://github.com/python-control/python-control/pull/868/commits/1d1c2890966ff80c606fd337e01e4bad9f75859b)), and for one or two cases I had to skip the test because `0.` would become `-0.` on some platforms, even after rounding.

Reply:
> Shouldn't this be `array` instead of `matrix`? And why skip the doctest?
> 
> This same issue come up multiple times below.

Doctest has an existing flaw/feature, in that it doesn't do float comparisons. Down below, it's just string compare.
When system precision affects the result, small tolerances can lead to test failure.
I'll have a hard look at it again, because it annoys me no end.

Reply:
Good point.  I have removed this and will go through an +SKIP any that fail.